### ISSUE TITLE
Ignoring per-file-ignores configuration

### ISSUE THREAD

**Issue by FieryDruid** [koxudaxi/ruff-pycharm-plugin#359]
**Describe the bug**

After updating to version 0.0.29, the plugin began to highlight rules that are ignored for specific files using the `per-file-ignores` section in `ruff.toml`. If ruff is run from the console with the same config, it does not find any errors.

**To Reproduce**
1. create package with some module in project
2. write any code that is recognized by ruff as an error
3. add this error code with path to this file in per-file-ignores
4. check this error highlighting

**Expected behavior**
Since ruff itself ignores the rule, the plugin should also not highlight it

**Screenshots**

Test code with error:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/ce4afa42-64a9-43b1-8896-a2339ef37613)

project structure with new folder `project_tests`:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/f8a43c78-8668-4fb2-8560-5fe1c7a47be0)

`per-file-ignores` section:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/49b4926b-163c-4acd-b4d6-a1beac81cf19)

`ruff check` resuilt from command line:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/d1d69173-3aa5-410b-9b24-d7b36fb9c3da)

**Environments (please complete the following information):**
 - IDE: `PyCharm Professional 2023.3.2`
 - OS: `Windows 10 + WSL (Ubuntu 22.04)`
 - Ruff Version: `0.1.9`
 - Plugin version: `0.0.29`

**Additional context**
It is likely that this may be related to the path to the config file in Windows + WSL: https://github.com/koxudaxi/ruff-pycharm-plugin/issues/358

but i fix this error by editing `ruff.xml` in `.idea/`. Everything worked fine on the previous version without any changes.

**Comment by Skorpyon** [comment]
Same thing.

**Comment by koxudaxi** [comment]
@FieryDruid 


I'm sorry for my late reply.
I tested your example. I can't reproduce the issue.

> It is likely that this may be related to the path to the config file in Windows + WSL: https://github.com/koxudaxi/ruff-pycharm-plugin/issues/358

Do you set your `ruff.toml` config path in the settings?

**Comment by FieryDruid** [comment]
> @FieryDruid
> 
> I'm sorry for my late reply. I tested your example. I can't reproduce the issue.
> 
> > It is likely that this may be related to the path to the config file in Windows + WSL: #358
> 
> Do you set your `ruff.toml` config path in the settings?

Yes, the config file is specified, but, due to the problem described in https://github.com/koxudaxi/ruff-pycharm-plugin/issues/358 I had to edit `ruff.xml` in `.idea` folder and specify a relative path in it

```xml
<option name="ruffConfigPath" value="ruff.toml" />
```
Could there be a problem due to such usage?


Still reproduced on this simple project:
[pythonProject.zip](https://github.com/koxudaxi/ruff-pycharm-plugin/files/14793335/pythonProject.zip)
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/d53d139e-dad2-4ae1-8375-3904a9950eab)


Pycharm: `2023.3.5`
Ruff: `0.1.9`
Plugin: `0.0.31`/`0.0.29`

And I tested this case on Ruff `0.3.4` with updated config:

```toml
[lint]
select = ["S101"]

[lint.per-file-ignores]
"tests/*" = ["S101"]
```

Plugin show me only assert:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/83bdec69-3e5e-4414-ad1b-f8c65e748d4b)

Ruff from CLI says:
![image](https://github.com/koxudaxi/ruff-pycharm-plugin/assets/36041629/aaafe117-ce3d-443c-bfc1-9fd07f954063)

**Comment by FieryDruid** [comment]
@koxudaxi 

Thanks for `0.0.33` release! 

Problem is actual with fixed https://github.com/koxudaxi/ruff-pycharm-plugin/issues/358

**Comment by koxudaxi** [comment]
@FieryDruid 
Thank you for checking it :)

**Comment by FieryDruid** [comment]
@koxudaxi 

sorry, it is not completed :(

I meant that error is still being reproduced after release

**Comment by koxudaxi** [comment]
@FieryDruid 
I tested the toml file from settings, which means it may not be taking relative paths into account. Could you test the absolute path?

**Comment by mihirsamdarshi** [comment]
Hi, This is happening for me as well.

However, I don't have a `ruff.toml` as the other bug reporters. I specify the per-file-ignores under `[tool.ruff.lint.per-file-ignores]` in `pyproject.toml`. 

I've had different issues with PyCharm before because this (newer) Python project is not in the root of my Pycharm workspace. Perhaps this could be a reason this isn't working for me?

**Comment by mihirsamdarshi** [comment]
I did some source code spelunking and I suspect that this might be an error with ruff/working directories when it gets the `--config` option?

I have two projects, one which is known good, and one which is known bad. The known bad one has the pyproject.toml file in a non-root project path.

I built and ran the plugin from scratch and enabled debug logging for the `com.intellij.execution` handler, which is where [`runRuff`](https://github.com/koxudaxi/ruff-pycharm-plugin/blob/c8b71aae743c497b6d53435d36107124f7ebae3e/src/com/koxudaxi/ruff/Ruff.kt#L371), then [`runCommand`](https://github.com/koxudaxi/ruff-pycharm-plugin/blob/c8b71aae743c497b6d53435d36107124f7ebae3e/src/com/koxudaxi/ruff/Ruff.kt#L221), then [`CapturingProcessHandler`](https://github.com/koxudaxi/ruff-pycharm-plugin/blob/c8b71aae743c497b6d53435d36107124f7ebae3e/src/com/koxudaxi/ruff/Ruff.kt#L248) end up.

In the good project, I noticed that the command being run was: `/Users/user/good_project/.venv/bin/ruff check --exit-zero --no-cache --force-exclude --no-fix --output-format json --stdin-filename tests/test__lib.py -`, and the working directory was `/Users/user/good_project`. The output when running this command manually in the terminal (with the venv activated) is the exact same as the one logged in the application.

And in the bad project the command was: `/Users/user/bad_project/server/.venv/bin/ruff check --exit-zero --no-cache --force-exclude --no-fix --output-format json --stdin-filename server/tests/test_app.py --config /Users/user/bad_project/server/pyproject.toml -`, and the working directory was `/Users/user/bad_project`. Again, the output when run manually is the same as seen the output seen in the log.

I believe the working directory of the command is the issue, because when I cd into `/Users/user/bad_project/server/` and run the exact same command manually, I see the expected output (no errors).

**Comment by ElipsisD** [comment]
For me, this bug is still an issue. 
I think it is due to the fact that I use WSL+Windows. After adding logs to the plugin, I noticed that the paths to files are incorrectly generated.

**Comment by ElipsisD** [comment]
![Image](https://github.com/user-attachments/assets/7820984b-89c2-41dd-b65a-12fa8d44bc15)

Using the same path to ruff. There are no errors when applied in the terminal. But ruff plugin shows errors by rules that are ignored. On the screenshot you can see that in the console in “filename” field the incorrect path to the file being checked is displayed.

**Comment by FieryDruid** [comment]
@koxudaxi fixed in `0.0.46`, thanks! 🎉
